### PR TITLE
Add sizes for input layer to support caffe to specify input dimensions

### PR DESCRIPTION
**Pull request by wzhongyuan**
## What changes were proposed in this pull request?

Add sizes for input layer to support caffe to specify input dimensions

## How was this patch tested?

unit test

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Input.scala [hunk 1]
```diff
@@ -36,7 +36,7 @@ import scala.reflect.ClassTag
  * @tparam T The numeric type in the criterion, usually which are [[Float]] or [[Double]]
  */
 @SerialVersionUID(- 8525406230282608924L)
-class Input[T: ClassTag]()(implicit ev: TensorNumeric[T])
+class Input[T: ClassTag](val sizes: Array[Int] = null)(implicit ev: TensorNumeric[T])
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Input.scala]: I suggest defining a new Shape class. We will use it in shape infer

Comment (yiheng) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Input.scala]: How do we deal with unfixed size, e.g. batch size dimension

**Comment by wzhongyuan** [comment]
@yiheng   yes, but we still need this PR , i will rebase and update

**Comment by wzhongyuan** [comment]
@yiheng   rebased and updated, and refined the test files to apply the change
PR validation passed http://172.168.2.101:8080/job/BigDL-PR-Validation-v3/90/

**Pull request closed without merging**